src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -132,6 +132,7 @@ class SecurityHandshaker : public Handshaker {
   RefCountedPtr<grpc_auth_context> auth_context_;
   tsi_handshaker_result* handshaker_result_ = nullptr;
   size_t max_frame_size_ = 0;
+  std::string tsi_handshake_error_;
```

Comment:
Can I know what will happen if TLS resumption happens? Will it re-use the the same `SecurityHandshaker` object, or create a new one? 
I don't think it will cause any problems here, as I think `SecurityHandshaker` will always be alive throughout the lifetime of  `tsi_handshaker_next`. Just asking a question off the top of my head while I was reading through the code :)

Reply:
SSL session caching and resumption is handled in the SSL TSI implementation.  It's not visible to the code here at all.